src/main/java/org/veupathdb/service/eda/ss/model/db/FilteredResultFactory.java

```diff
@@ -172,10 +174,14 @@ public static void produceTabularSubsetFromFile(Study study, Entity outputEntity
       final Iterator<Long> idIndexStream = driver.reduce(dataFlowTree);
 
       // Open streams of output variables and ancestors identifiers used to decorate ID index stream to produce tabular records.
-      final List<Iterator<VariableValueIdPair<String>>> outputVarStreams = outputVariables.stream()
-          .map(Functions.fSwallow(
-              outputVariable -> binaryValuesStreamer.streamIdValuePairs(study, (VariableWithValues<?>) outputVariable, reportConfig)))
-          .collect(Collectors.toList());
+      List<FormattedTabularRecordStreamer.ValueStream<String>> outputVarStreams = new ArrayList<>();
+      for (Variable outputVar: outputVariables) {
+        VariableWithValues<?> varWithVals = (VariableWithValues<?>) outputVar;
```

Comment:
i know it is preexisting code, but why isn't the outputVariables param typed as `VariableWithValues<?>`.  ie, why cast?

Reply:
Yeah, I agree. I can make this change, though I think there are a quite a few places where we pass around Variable where we'd always want a VarWithValues.